source/frontend/src/features/leases/detail/LeasePages/stakeholders/AddLeaseStakeholderContainer.tsx

```diff
@@ -127,14 +127,13 @@ export const AddLeaseStakeholderContainer: React.FunctionComponent<
             op.person = matchingPerson;
           }
         });
-        stakeholder.stakeholderType = stakeholder?.stakeholderType
-          ? stakeholder.stakeholderType
-          : isPayableLease
-          ? 'OWNER'
-          : 'TEN';
         return stakeholder;
       }) ?? [];
-    const formTenants = tenantsWithPersons?.map(t => new FormStakeholder(undefined, t)) ?? [];
+    const stakeholderType = isPayableLease ? 'OWNER' : 'TEN';
```

Comment:
I think this may cause issues if a user has owners/tenants and then swaps the lease from payable to not-payable.

Just a note - let's be careful about the scope of these "refactors" - if we don't have existing tests, and the refactor doesn't include tests to validate the before/after functionality, than these are code changes and not refactors, since we are altering functionality. That is potentially fine - but we need to make sure that the FT team is aware of the scope of these changes - otherwise when this is merged the FT team will not know what to re-test. (this story is psp-8905, which talks about contact summary model changes, I don't think it would be obvious to the FT team that lease stakeholder functionality would be affected).

Reply:
I agree that refactors should be minimal. In this case, the refactor exposed an improper use of a common model for something that is quite specific (the lease tenants). In an ideal world, the code would have been separated making the changes here much smaller. Fixing this is part of maintaining the code base and cleaning shortcuts that where taken at the time (tech debt). Also, all code changes introduce some risk into the application, and no changes make the application harder to enhance in the future, so there is always a tradeoff.

Reply:
Fair enough, the point remains about the testing team, please make sure the scope of the changes included in this story are clear.